src/modalities/utils/mfu.py

```diff
@@ -38,47 +42,47 @@ def _get_theoretical_gpu_peak_performance_single(precision: torch.dtype, gpu_typ
 
 def get_theoretical_gpu_peak_performance(model: FSDP, world_size: int) -> Optional[Number]:
     """
-    Calculates the accummulated theoretical peak performance based on all GPUs, i.e.,
+    Calculates the accumulated theoretical peak performance based on all GPUs, i.e.,
       #GPU=world_size, in units FLOPs / s for given gpu type.
 
     Args:
         model (FSDP): The model for which to calculate the theoretical peak performance.
         world_size (int): The number of GPUs used in parallel.
 
     Returns:
-        (Number, optional): The accummulated theoretical peak performance of all GPUs,
+        (Number, optional): The accumulated theoretical peak performance of all GPUs,
           or None if it cannot be calculated.
     """
-    # BUG: this code does not work due to model not having mixed_precision attribute!
-    # a good implementation can be found here:
-    # https://github.com/pytorch/torchtitan/blob/b291ad662493b63d25b038a30a915082d3617baf/torchtitan/train.py#L80
-
-    # if torch.cuda.is_available() and torch.cuda.device_count() > 0:
-    #     precision = model.mixed_precision.param_dtype
-    #     if model.mixed_precision.reduce_dtype != precision or model.mixed_precision.buffer_dtype != precision:
-    #         warnings.warn(f"Could not get theoretical GPU peak performance for mixed precision type = {precision}.")
-    #         return None
-    #     else:
-    #         device_name = torch.cuda.get_device_name()
-    #         if device_name.startswith("NVIDIA A100"):
-    #             single_gpu_peak_performance = _get_theoretical_gpu_peak_performance_single(precision, "A100")
-    #         elif device_name.startswith("NVIDIA H100"):
-    #             single_gpu_peak_performance = _get_theoretical_gpu_peak_performance_single(precision, "H100")
-    #         else:
-    #             warnings.warn(f"Could not get theoretical GPU peak performance for unknown device = {device_name}.")
-    #             return None
-    #         if single_gpu_peak_performance is None:
-    #             warnings.warn(
-    #                 f"Could not get theoretical GPU peak performance for device = {device_name} "
-    #                 f"and mixed precision type = {precision}."
-    #             )
-    #             return None
-    #         else:
-    #             return single_gpu_peak_performance * world_size
-    # else:
-    #     return None
-
-    return None
+    if torch.cuda.is_available() and torch.cuda.device_count() > 0:
+        if isinstance(model, FSDP1):
+            precision = model.mixed_precision.param_dtype
+            if model.mixed_precision.reduce_dtype != precision or model.mixed_precision.buffer_dtype != precision:
+                warnings.warn(f"Could not get theoretical GPU peak performance for mixed precision type = {precision}.")
```

Comment:
Isn't this warning misleading? 
From the if statement I would have expected a warning such as:
"Could not get theoretical GPU peak performance. In mixed precision, the param_dtype must be equal to reduce_dtype and buffer_dtype."